EFCore.VisualBasic/Scaffolding/Internal/VisualBasicDbContextGenerator.tt

```diff
@@ -0,0 +1,307 @@
+﻿<#@ template language="VB" hostSpecific="true" linePragmas="false" #>
+<#@ assembly name="Microsoft.EntityFrameworkCore" #>
+<#@ assembly name="Microsoft.EntityFrameworkCore.Design" #>
+<#@ assembly name="Microsoft.EntityFrameworkCore.Relational" #>
+<#@ assembly name="Microsoft.Extensions.DependencyInjection.Abstractions" #>
+<#@ parameter name="Model" type="Microsoft.EntityFrameworkCore.Metadata.IModel" #>
+<#@ parameter name="Options" type="Microsoft.EntityFrameworkCore.Scaffolding.ModelCodeGenerationOptions" #>
+<#@ parameter name="NamespaceHint" type="System.String" #>
+<#@ import namespace="System.Collections.Generic" #>
+<#@ import namespace="System.Linq" #>
+<#@ import namespace="System.Text" #>
+<#@ import namespace="Microsoft.EntityFrameworkCore" #>
+<#@ import namespace="Microsoft.EntityFrameworkCore.Design" #>
+<#@ import namespace="Microsoft.EntityFrameworkCore.Scaffolding" #>
+<#@ import namespace="Microsoft.Extensions.DependencyInjection" #>
+<#@ import namespace="EntityFrameworkCore.VisualBasic.Design" #>
+<#
+    Dim services = DirectCast(Host, IServiceProvider)
+    Dim providerCode = services.GetRequiredService(Of IProviderConfigurationCodeGenerator)()
+    Dim annotationCodeGenerator = services.GetRequiredService(Of IAnnotationCodeGenerator)()
+    Dim code = services.GetRequiredService(Of IVisualBasicHelper)()
+
+    Dim importsList = New List(Of String) From {
+        "System",
+        "System.Collections.Generic",
+        "Microsoft.EntityFrameworkCore"
+    }
+
+    If NamespaceHint <> Options.ModelNamespace AndAlso
+       Not String.IsNullOrEmpty(Options.ModelNamespace) Then
+        importsList.Add(Options.ModelNamespace)
+    End If
+
+    If Not string.IsNullOrEmpty(NamespaceHint) Then
+#>
+Namespace <#= NamespaceHint #>
```

Comment:
Yes it's true, I still have to take into account the specificities of VB for namespaces.
The currently failing unit tests are related to this.

I could use the `Global` keyword to set the namespace outside of the root namespace all the time.
This is currently done for migration files (I think, `MigrationsCodeGenerator` does not provide the project root namespace).

But, since the project's root namespace is provided to the template, I'd rather try to improve the user experience and create a helper method to try to determine the desired namespace to use.